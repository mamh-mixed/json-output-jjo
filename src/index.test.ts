import { parse } from './index';

describe('jjo', () => {
    it('parses standard types', () => {
        expect(parse([
            'string=string',
            'number=1234',
            'boolean=true',
            'float=1.234',
            'empty=',
            'badfloat=1.123.4'
        ])).toMatchSnapshot()
    });

    it('parses json types', () => {
        expect(parse([
            'obj={"key": "value", "flag": false}',
            'arr=[{"something": "else", "num": 123}, true, "string"]'
        ])).toMatchSnapshot()
    });

   xit('handles invalid json', () => {
        expect(parse([
            'obj={"key": "value", "flag: false}',
            'arr=[{"something": "else" "num": 123}, true, "string"]'
        ])).toMatchSnapshot()
    });

    it('parses files', () => {
        expect(parse([
            `obj=:${__dirname}/__fixtures__/data.json`
        ])).toMatchSnapshot()
    });

    it('parses files literally', () => {
        expect(parse([
            `obj=@${__dirname}/__fixtures__/data.json`
        ])).toMatchSnapshot()
    });

    xit('handles files with invalid json', () => {
        expect(parse([
            `obj=:${__dirname}/__fixtures__/bad_data.json`
        ])).toMatchSnapshot()
    });
});