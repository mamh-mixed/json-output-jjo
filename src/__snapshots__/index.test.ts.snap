// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jjo handles files with invalid json 1`] = `
Object {
  "obj": Object {
    "boolean": true,
    "key": "value",
    "number": 123.12,
  },
}
`;

exports[`jjo parses files 1`] = `
Object {
  "obj": Object {
    "boolean": true,
    "key": "value",
    "number": 123.12,
  },
}
`;

exports[`jjo parses files literally 1`] = `
Object {
  "obj": "{
    \\"boolean\\": true,
    \\"key\\": \\"value\\",
    \\"number\\": 123.12
}",
}
`;

exports[`jjo parses json types 1`] = `
Object {
  "arr": Array [
    Object {
      "num": 123,
      "something": "else",
    },
    true,
    "string",
  ],
  "obj": Object {
    "flag": false,
    "key": "value",
  },
}
`;

exports[`jjo parses standard types 1`] = `
Object {
  "badfloat": "1.123.4",
  "boolean": true,
  "boolean2": false,
  "empty": "",
  "float": 1.234,
  "number": 1234,
  "string": "string",
}
`;
